src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java

```diff
@@ -102,12 +103,35 @@ public void save( final URI identifier, final T element ) throws IOException {
 		}
 	}
 
-	@SuppressWarnings( "unchecked" )
+	/**
+	 * You need to maintain the stream and are responsible for closing it properly.
+	 */
+	public void save( final OutputStream outputStream, final T element ) throws IOException {
+		final Output output = new Output( new LZ4FrameOutputStream( outputStream ) );
+		output.writeString( version );
+		kryo.writeClassAndObject( output, element );
+		// Flush the stream to avoid corrupt file data
+		output.flush();
```

Comment:
I thought we discussed this as try-with-resources? Just to close the outer stream at all times.
```
try (final Output output = new Output( new LZ4FrameOutputStream( outputStream ) ) ) {
    output.writeString( version );
    kryo.writeClassAndObject( output, element );
}
```